File: api/models/ResultQuiz.js
const mongoose = require('mongoose');

const resultQuizSchema = new mongoose.Schema({
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true,
    },
    quizId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Quiz",
        required: true,
    },
    score: {
        type: String,
        required: true,
    },
    testAt: { 
        type: Date, 
        default: Date.now 
    }
    
})

const ResultQuiz = mongoose.models.ResultQuiz || mongoose.model('ResultQuiz', resultQuizSchema);

module.exports = ResultQuiz;

File: app/lib/models/resultquiz.dart
class ResultQuiz {
  String? id;
  String? quizId;
  String? userId;
  String? testAt;
  String? score;

  ResultQuiz({
    this.id,
    this.quizId,
    this.userId,
    this.score,
    this.testAt,
  });

  ResultQuiz.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    quizId = json['quizId'];
    userId = json['userId'];
    score = json['score'];
    testAt = json['testAt'];
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'quizId': quizId,
      'userId': userId,
      'score': score,
      'testAt': testAt,
    };
  }
}
